core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java

```diff
@@ -13,6 +15,7 @@
 public class NativeImageBuildRemoteContainerRunner extends NativeImageBuildContainerRunner {
 
     private static final Logger log = Logger.getLogger(NativeImageBuildRemoteContainerRunner.class);
+    private static final String CONTAINER_BUILD_VOLUME_NAME = "quarkus-native-builder-image-project-volume";
```

Comment:
Ideally I would prefer a dynamically generated unique volume name, but I am not sure it's worth the effort.
We could also do something like `fixed-name-YYYY-MM-DD-HHMM` to avoid potential conflicts with existing volumes.
WDYT?

Reply:
It's straightforward to let docker generate the unique volume id, I quickly tested it locally:
```
final String[] createVolumeCommand = new String[] { containerRuntime.getExecutableName(), "volume", "create" };
volumeId = runCommandAndReadOutput(createVolumeCommand, "Failed to create temp volume.");
```
Using this as a first step in `preBuild` and replacing all `CONTAINER_BUILD_VOLUME_NAME` with `volumeId`.

Reply:
Unfortunately it's not that simple. If you manually create the volume before starting the container, the volume (when mount) is owned by `root:root` and `native-image` fails to write to it in subsequent steps.

## Creating the volume explicitly
```
$ podman volume create test-volume                         
test-volume
$ podman create --name temp -v test-volume:/project quay.io/quarkus/ubi-quarkus-native-image:21.0.0-java11                         
bfa2406d5fa0a003af99c91f077ed989006eb2441004135773429c425f9ce997
$ podman run --rm --entrypoint /bin/bash -it -v test-volume:/project quay.io/quarkus/ubi-quarkus-native-image:21.0.0-java11 -c "ls -la"
total 0
drwxr-xr-x.  2 root root 6 Mar  3 11:40 .
drwxr-xr-x. 19 root root 6 Mar  3 11:41 ..
```

## Creating the volume implicitly
```
$ podman create --name temp -v test-volume:/project quay.io/quarkus/ubi-quarkus-native-image:21.0.0-java11
e30ffb76aa320448a3e645d7030c4dcf76b0efb83b68e3bc58a2e655c683d1de
$ podman run --rm --entrypoint /bin/bash -it -v test-volume:/project quay.io/quarkus/ubi-quarkus-native-image:21.0.0-java11 -c "ls -la"
total 0
drwxr-xr-x.  2 quarkus quarkus 6 Mar  3 11:42 .
drwxr-xr-x. 19 root    root    6 Mar  3 11:42 ..
```

I am not a containers expert though so I'll be happy to be corrected :)

Reply:
Interesting! I checked the following:

- docker 19.03.13 on Ubuntu 20.10
- docker 20.10.2 on Ubuntu 20.10 in WSL
- podman 2.0.6 on Ubuntu 20.10
- podman 1.6.4 on CentOS 7.9
- podman 3.0.1 on CentOS 7.9

in all but the last case, the `/project` folder was owned by the `quarkus` user also when creating the volume _explicitly_. So something seems to have changed for podman 3. I'm guessing you're running on podman 3 as well?

Reply:
Indeed I am using podman 3.0.1 on Fedora 33.

Thanks for the additional info @jonathan-meier, I created https://github.com/containers/podman/issues/9608.